### PR TITLE
chore(tech debt): testing components in CustomerAppeal

### PR DESCRIPTION
**Pull request by andrehadianto**
[Notion link](notion-link-here) 

This PR adds... 
- components testing for Customer Appeal directory

- [x] I've kept this PR as small as possible (~600 lines) by splitting it into PRs with manageable chunks of code
- [x] I've requested reviews from 1-2 reviewers
- [ ] I've added [accessibility IDs](https://www.notion.so/dc5f4ce910f7431c84344ac79344e9f5?v=d487366816834dd4ab8dc12e0b5928f3) to my components
- [ ] I've added/updated unit/integration tests
- [ ] I've added/updated [e2e tests](https://www.notion.so/e2e-Documentation-a096d3e0bf75485e85ad692af8371ef3) for at least the happy flow
- [ ] I've added [Chinese translations for i18n setup](https://www.notion.so/Translations-Dev-Guide-8c1da838982a4f59a386ec96ac6780c8)
- [ ] I've added documentation in README/[Notion](https://www.notion.so/82b92fb1007640328dab9582c0a3694e?v=3b6ce48202cf40ad8450553799b13146)

### CODE REVIEW

File: src/components/CustomerAppeal/ReasonSelection/ReasonSelectionHeader.test.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+describe("ReasonSelectionHeader", () => {
+  afterEach(() => {
+    cleanup();
+    jest.resetAllMocks();
+  });
+
+  it("should render the header correctly", async () => {
+    expect.assertions(1);
+    const { queryByText } = render(
+      <ReasonSelectionHeader title="Reason Header" />
```
Comment (maksha19) [inline src/components/CustomerAppeal/ReasonSelection/ReasonSelectionHeader.test.tsx:15]: just curious, what happens if the title with empty string

**Review by maksha19 — CHANGES_REQUESTED** [review]

File: src/components/CustomerAppeal/ReasonSelection/ReasonSelectionCard.test.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+const onCancel = jest.fn();
+const onReasonSelection = jest.fn();
+const getReasons = jest
+  .fn()
+  .mockReturnValue([
+    { description: "reason-a" },
+    { description: "reason-b" },
+    { description: "reason-c" },
+  ]);
```
[the remark this answers is not in the document]
Comment (andrehadianto) [reply src/components/CustomerAppeal/ReasonSelection/ReasonSelectionCard.test.tsx:14]: this one is to simulate the fact that `getReasons` is a function in `ReasonSelectionCard.tsx`, so I think it's better to also make it a `fn()`

File: src/components/CustomerAppeal/ReasonSelection/ReasonSelectionHeader.test.tsx (hunk 1 above)
Comment (andrehadianto) [reply src/components/CustomerAppeal/ReasonSelection/ReasonSelectionHeader.test.tsx:15]: The header DOM will have `""` as the value

Comment (maksha19) [reply src/components/CustomerAppeal/ReasonSelection/ReasonSelectionHeader.test.tsx:15]: thanks.. so the `expect(...)` is going to null.. can include that also

File: src/components/CustomerAppeal/ReasonSelection/ReasonSelectionCard.test.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrehadianto) [reply src/components/CustomerAppeal/ReasonSelection/ReasonSelectionCard.test.tsx]: As per discussed, will change `getReasons` to mock by its component's parameter type => `mockReasons: Reason[]`

File: src/components/CustomerAppeal/CustomerAppealScreen.test.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        features: {
+          ...defaultFeatures,
+          id: {
+            type: "STRING",
+            scannerType: "QR",
+            validation: "NRIC",
+          },
+          campaignName: "Some Campaign Name",
+        },
+        policies: [
```
[the remark this answers is not in the document]
Comment (maksha19) [reply src/components/CustomerAppeal/CustomerAppealScreen.test.tsx:41]: clarified on call 🤝

**Review by maksha19 — APPROVED** [review]
LGTM 🚀

**Pull request merged**